### PR TITLE
Fix SegFault on SegmentURL handling

### PR DESCRIPTION
**Pull request by MisterD81**
While trying to create a Kodi addon for Rakuten I got several times a segmentation fault for some videos. It turned out that the MPD of the crashing ones was different then for the working ones.
In the crashing ones they use in SegmentURL elements both "media" and "mediaRange" attribute at the same time. 
Therefore I adjusted Segment structure to keep both attributes at the same time. In addition to that the range header handling was added for that case too.

Additional side-effect is that it fixes this issue with the Youtube addon:
https://github.com/jdf76/plugin.video.youtube/issues/595

I don't know if those audio streams are using the same structure like Rakuten but at least no I have sound there as well.

### CODE REVIEW

**Comment by peak3d** [comment]
@MisterD81 can you pls. attach the mpd? In general different mpd representations should be separated by flag (if possible) to reduce amount of memory kept by internal segment list.

**Comment by MisterD81** [comment]
Sure. Please find attached one of the crashing MPD files:
[Sample-MPD.txt](https://github.com/peak3d/inputstream.adaptive/files/3132070/Sample-MPD.txt)

**Comment by MisterD81** [comment]
@peak3d : Do you need further information for merging this PR or should I rework parts? Please just let me know. I really would like to see this integrated as it prevents Kodi from crashing when trying to play such streams.

**Comment by peak3d** [comment]
PR makes sense looking at the manifest, I'll take a final look on in tomorrow

File: src/common/AdaptiveStream.cpp [hunk 1]
```diff
@@ -268,15 +268,13 @@ bool AdaptiveStream::prepareDownload(const AdaptiveTree::Segment *seg)
...
-        if (~seg->range_end_)
-          sprintf(rangebuf, "bytes=%" PRIu64 "-%" PRIu64, seg->range_begin_ + fileOffset, seg->range_end_ + fileOffset);
-        else
-          sprintf(rangebuf, "bytes=%" PRIu64 "-", seg->range_begin_ + fileOffset);
-        rangeHeader = rangebuf;
-      }
+      uint64_t fileOffset = seg != &current_rep_->initialization_ ? m_segmentFileOffset : 0;
+      if (~seg->range_end_)
+        sprintf(rangebuf, "bytes=%" PRIu64 "-%" PRIu64, seg->range_begin_ + fileOffset, seg->range_end_ + fileOffset);
+      else
```
Comment (peak3d) [inline src/common/AdaptiveStream.cpp]: This does not work for URLSegment representations with no byte range given. The complete block must be placed inside a condition if (~seg->range_begin_)

Comment (MisterD81) [reply src/common/AdaptiveStream.cpp]: @peak3d : For sure this could happen. I added the requested changes.

**Pull request merged**